lib/src/widgets/custom_floating_action_button.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:exerlog/src/widgets/theme/theme_provider.dart';
+import 'package:flutter/material.dart';
+
+class CustomFloatingActionButton extends StatelessWidget {
+  final IconData? icon;
+  final double? size;
+  final VoidCallback? onTap;
+
+  CustomFloatingActionButton({
+    @required this.icon,
+    this.size = 50.0,
+    @required this.onTap,
+  });
+
+  @override
+  Widget build(BuildContext context) => ThemeProvider(
+        builder: (context, theme) => Container(
+          decoration: BoxDecoration(
+            color: theme.colorTheme.primaryColor,
+            borderRadius: BorderRadius.circular(size!),
```

Comment:
Seeing as you gave size a default value, you can just not make it nullable. Then you don't have to force it here